soc/arm/nordic_nrf/common/CMakeLists.txt

```diff
@@ -2,7 +2,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_NRF soc_nrf_common.S)
-zephyr_library_sources_ifdef(CONFIG_POWEROFF poweroff.c)
```

Comment:
This `poweroff.c` file should be also removed.

But why not instead of reintroducing those separate `poweroff.c` filse just replace: https://github.com/zephyrproject-rtos/zephyr/blob/959e0e829dd17b55c02b17535fe58765d3339454/soc/arm/nordic_nrf/common/poweroff.c#L13
with something like:
```
#if defined(SOC_SERIES_NRF51X) || defined(SOC_SERIES_NRF52X)
	nrf_power_system_off(NRF_POWER);
#else
	nrf_regulators_system_off(NRF_REGULATORS);
#endif
```
and include proper headers with similar construct?